### PR TITLE
[fix] stream field in llmconfig not work

### PR DESCRIPTION
**Pull request by Wei-Jianan**
### **User description**
**Fix**
-- https://github.com/geekan/MetaGPT/issues/1257
**Bug description**
~/.metagpt/config2.yaml
```
llm:  
  stream: False
```
does not affect the way to call llm.aask.

___

### **PR Type**
Bug fix, Enhancement

___

### **Description**
- Integrated the `stream` configuration from `config2.yaml` into the `aask` method to respect the user's settings.
- Minor formatting improvements in benchmark and ranker factory files to enhance code readability and maintainability.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>base_llm.py</strong><dd><code>Integrate stream configuration in aask method</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
metagpt/provider/base_llm.py

<li>Imported <code>config</code> from <code>metagpt.config2</code>.<br> <li> Modified the default value of the <code>stream</code> parameter in the <code>aask</code> method <br>to be <code>None</code>.<br> <li> Added a conditional to set <code>stream</code> based on the <code>llm.stream</code> <br>configuration if <code>stream</code> is <code>None</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/geekan/MetaGPT/pull/1258/files#diff-7b80d6287bf4cf75c4061b77e9af5c5e017ba6d774301fa43823e6997c6b423f">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ranker.py</strong><dd><code>Refactor and improve import statements in ranker factory</code>&nbsp; </dd></summary>
<hr>
      
metagpt/rag/factories/ranker.py

<li>Reordered import statements for ranker configurations.<br> <li> Improved import formatting for <code>FlagEmbeddingReranker</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/geekan/MetaGPT/pull/1258/files#diff-ce587548bf78ff1dc92d4bc991ffce871cce1384ee4d32bd0f0580cfecf297c2">+6/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Formatting
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>base.py</strong><dd><code>Minor formatting cleanup</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
metagpt/rag/benchmark/base.py

- Minor formatting change by removing extra whitespace.

</details>
    

  </td>
  <td><a href="https://github.com/geekan/MetaGPT/pull/1258/files#diff-434f146a1d34a86b05c5be1ad008bffb52abc747dc467a91cef37b29bd96c836">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: metagpt/provider/base_llm.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
     wait_random_exponential,
 )
 
+from metagpt.config2 import config
```
Comment (geekan) [inline metagpt/provider/base_llm.py:25]: Thank you very much for your contribution.

The current design does not introduce global config in base_llm, but passes it in from the outside.

File: metagpt/rag/benchmark/base.py [hunk 2]
```diff
@@ -121,7 +121,7 @@ def mean_reciprocal_rank(self, nodes: list[NodeWithScore], reference_docs: list[
                     return mrr_sum
 
         return mrr_sum
-        
+
```
Comment (geekan) [inline metagpt/rag/benchmark/base.py:124]: Text reformat should be separated from bugfix as much as possible

File: metagpt/provider/base_llm.py (hunk 1 above)
Comment (Wei-Jianan) [reply metagpt/provider/base_llm.py]: thank you for pointing that out.
I have changed the code by using BaseLLM.config.stream other than from global.

File: metagpt/rag/benchmark/base.py (hunk 2 above)
Comment (Wei-Jianan) [reply metagpt/rag/benchmark/base.py:124]: The reformatting changes are automatically applied by running the command 'pre-commit run --all-files', which is introduced in the contribution guide. Should I revert those changes? 吴老板我听您的。

File: metagpt/provider/base_llm.py [hunk 3]
```diff
@@ -146,6 +146,8 @@ async def aask(
             message.append(self._user_msg(msg, images=images))
         else:
             message.extend(msg)
+        if stream is None:
+            stream = self.config.stream
```
Comment (geekan) [inline metagpt/provider/base_llm.py:150]: If modified in this way, the default stream will also become False due to the setting in llm_config. I recommend changing the setting in llm_config to the default True to maintain consistency

Comment (Wei-Jianan) [reply metagpt/provider/base_llm.py:150]: Oh Im sorry. I should be more carefully。I should have thought of that。

**Review by geekan — APPROVED** [review]

**Pull request merged**